src/bottom-bar/use-on-complete-callback.js

```diff
@@ -135,11 +139,26 @@ export default function useOnCompleteCallback() {
         useOnCompleteWhenValidNotRequiredClick()
     const onCompleteWithoutValidationClick =
         useOnCompleteWithoutValidationClick()
+    const hasCompulsoryDataElementOperandsToFillOut =
+        useHasCompulsoryDataElementOperandsToFillOut()
+    const setCompleteAttempted = useEntryFormStore(
+        (state) => state.setCompleteAttempted
+    )
 
     return () => {
         let promise
 
-        if (isLoading && offline) {
+        if (hasCompulsoryDataElementOperandsToFillOut) {
+            setCompleteAttempted(true)
+            cancelCompletionMutation()
```

Comment:
I think that doing ```       completeStatus: {
                    ...previousDataValueSet.completeStatus,
                    complete: !completed,
                },``` in cancelCompletionMutation messes up the complete status a bit. In you try to mark as complete successfully, then mark as incomplete successfully and then finally mark as complete unsuccessfully you will see the complete status = true even if the completion did fail. maybe we just want to set it as ```       completeStatus: {
                    ...previousDataValueSet.completeStatus,
                    complete: false,
                },``` ?